state/txns.go

```diff
@@ -84,6 +88,12 @@ func (r *multiModelRunner) Run(transactions jujutxn.TransactionSource) error {
 		if err != nil {
 			return nil, errors.Trace(err)
 		}
+
+		if txnOpLogger.IsTraceEnabled() {
+			txnOpLogger.Tracef(string(debug.Stack()))
+			txnOpLogger.Tracef(
+				strings.Join(transform.Slice(ops, func(op txn.Op) string { return fmt.Sprintf("%#v", op) }), "\n"))
```

Comment:
`pretty.Sprint(op)` would be more verbose but the formatting is visually much more grokable.

Reply:
Having spat out such logs to debug a prior issue, the in-lining is actually desirable for comparing the list contents.

The formatting looks better for a single one, true, but not so much in bulk.